#!/usr/bin/env python
PACKAGE = "collision_avoidance"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#           Name                      Type       Level  Description                                     Default       Min     Max
gen.add("robot_frame",                str_t,       1,   "The robots frame",                             "base_link")
gen.add("robot_height",               double_t,    1,   "Height of the robot in meter",                 0.5,          0,      5)

gen.add("radius",                     double_t,    2,   "Radius of the robot in meter",                 0.41,         0,      1)
gen.add("security_distance",          double_t,    2,   "Security distance in meter",                   0.61,         0,      2)
gen.add("epsilon",                    double_t,    2,   "Epsilon in ORM",                               0.1,          0,      1)

gen.add("min_distance_hold",          double_t,    3,   "Obstacles considered when no input in meter",  0.61,         0,      10)

gen.add("min_change_in_direction",    double_t,    4,   "min_change_in_direction in ORMT in degrees",   5.0,          -180,   180)
gen.add("max_change_in_direction",    double_t,    4,   "max_change_in_direction in ORMT in degrees",   45.0,         -180,   180)

gen.add("min_opposite_direction",     double_t,    5,   "min_opposite_direction in ORMT in degrees",    0.0,          -180,   180)
gen.add("max_opposite_direction",     double_t,    5,   "max_opposite_direction in ORMT in degrees",    10.0,         -180,   180)

gen.add("max_data_age",               double_t,    6,   "max_data_age in ORMT in seconds",              1.0,          0,      10)
gen.add("polar_size",                 int_t,       6,   "polar_size in ORMT",                           360,          0,      3600)

gen.add("max_xy_vel",                 double_t,    7,   "Maximum horizontal velocity (m/s)",            2.0,          0,      10)
gen.add("max_z_vel",                  double_t,    7,   "Maximum vertical velocity (m/s)",              0.8,          0,      10)
gen.add("max_yaw_rate",               double_t,    7,   "Maximum yaw_rate velocity (deg/s)",            70,           0,      360)

exit(gen.generate(PACKAGE, "collision_avoidance", "CollisionAvoidance"))
